Source/WTF/wtf/glib/SysprofAnnotator.h

```diff
@@ -0,0 +1,436 @@
+        case WakeUpAndApplyDisplayListStart:
+            beginMark(nullptr, tracePointCodeName(code).spanIncludingNullTerminator(), "%s", "");
+            break;
+
+        case VMEntryScopeEnd:
+        case WebAssemblyCompileEnd:
+        case WebAssemblyExecuteEnd:
+        case DumpJITMemoryStop:
+        case FromJSStop:
+        case IncrementalSweepEnd:
+        case MainResourceLoadDidEnd:
+        case SubresourceLoadDidEnd:
+        case FetchCookiesEnd:
+        case StyleRecalcEnd:
+        case RenderTreeBuildEnd:
+        case PerformLayoutEnd:
+        case PaintLayerEnd:
+        case AsyncImageDecodeEnd:
+        case RAFCallbackEnd:
+        case MemoryPressureHandlerEnd:
+        case UpdateTouchRegionsEnd:
+        case DisplayListRecordEnd:
+        case ComputeEventRegionsEnd:
+        case RenderingUpdateEnd:
+        case CompositingUpdateEnd:
+        case DispatchTouchEventsEnd:
+        case ParseHTMLEnd:
+        case DisplayListReplayEnd:
+        case ScrollingThreadRenderUpdateSyncEnd:
+        case ScrollingThreadDisplayDidRefreshEnd:
+        case RenderTreeLayoutEnd:
+        case PerformOpportunisticallyScheduledTasksEnd:
+        case WebXRLayerStartFrameEnd:
+        case WebXRLayerEndFrameEnd:
+        case WebXRSessionFrameCallbacksEnd:
+        case WebHTMLViewPaintEnd:
+        case BackingStoreFlushEnd:
+        case BuildTransactionEnd:
+        case SyncMessageEnd:
+        case SyncTouchEventEnd:
+        case InitializeWebProcessEnd:
+        case RenderingUpdateRunLoopObserverEnd:
+        case LayerTreeFreezeEnd:
+        case FlushRemoteImageBufferEnd:
+        case CreateInjectedBundleEnd:
+        case PaintSnapshotEnd:
+        case RenderServerSnapshotEnd:
+        case TakeSnapshotEnd:
+        case SyntheticMomentumEnd:
+        case CommitLayerTreeEnd:
+        case ProcessLaunchEnd:
+        case InitializeSandboxEnd:
+        case WebXRCPFrameWaitEnd:
+        case WebXRCPFrameStartSubmissionEnd:
+        case WebXRCPFrameEndSubmissionEnd:
+        case WakeUpAndApplyDisplayListEnd:
+            endMark(nullptr, tracePointCodeName(code).spanIncludingNullTerminator(), "%s", "");
+            break;
+
+        case DisplayRefreshDispatchingToMainThread:
+        case ScheduleRenderingUpdate:
+        case TriggerRenderingUpdate:
+        case ScrollingTreeDisplayDidRefresh:
+        case SyntheticMomentumEvent:
+        case RemoteLayerTreeScheduleRenderingUpdate:
+            instantMark(tracePointCodeName(code).spanIncludingNullTerminator(), "%s", "");
+            break;
+
+        case WTFRange:
+        case JavaScriptRange:
+        case WebCoreRange:
+        case WebKitRange:
+        case WebKit2Range:
+        case UIProcessRange:
+        case GPUProcessRange:
+            break;
+        }
+    }
+
+    static void initialize(ASCIILiteral processName)
+    {
+        singleton().m_processName = processName;
+        sysprof_collector_init();
```

Comment:
So, if we take the approach of calling `construct()` from here, we can make process name a construct parameter and move `sysprof_collector_init()` to the constructor too.